test/quoteSpec.js

```diff
@@ -0,0 +1,240 @@
+      assert.equal(response, undefined)
+    })
+
+    it('should return undefined on a timeout', function * () {
+      this.plugin.sendMessage = () => Promise.resolve(null)
+      this.params.timeout = 10
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.equal(response, undefined)
+    })
+
+    describe('quoteByPacket', function () {
+      it('should parse quote params from packet', function * () {
+        // the response we're using gives sourceExpiryDuration of 0
+        this.result.expiresAt = moment().format()
+
+        const response = yield ILQP.quoteByPacket(
+          this.plugin,
+          Packet.serialize({
+            amount: '1',
+            address: 'test.local.bob'
+          })
+        )
+        assert.deepEqual(
+          response,
+          this.result)
+      })
+    })
+  })
+
+  describe('_getQuote', function () {
+    beforeEach(function () {
+      this.params = {
+        plugin: this.plugin
+      }
+    })
+
+    it('should return the data from the message response', function * () {
+      this.plugin.sendMessage = (msg) => {
+        this.response.data.id = msg.data.id
+        this.plugin.emit('incoming_message', this.response)
+        return Promise.resolve(null)
+      }
+
+      const response = yield ILQP._getQuote(this.params)
+      assert.deepEqual(
+        response,
+        this.response.data.data)
+    })
+
+    it('should resolve to undefined on an error', function * () {
+      this.params.timeout = 10
+      const response = yield ILQP._getQuote(this.params)
+
+      assert.equal(response, undefined)
+    })
+  })
+
+  describe('_sendAndReceiveMessage', function () {
+    beforeEach(function () {
+      this.params = {
+        plugin: this.plugin, 
+        method: 'quote_response',
+        message: {
+          data: { id: this.id }
+        }
+      }
+    })
+
+    it('should resolve on response message', function * () {
+      const promise = ILQP._sendAndReceiveMessage(this.params)
+      this.plugin.emit('incoming_message', this.response)
+      yield promise
+    })
+
+    it('should ignore a message that doesn\'t match its id', function * () {
+      const promise = ILQP._sendAndReceiveMessage(this.params)
+      
+      this.plugin.emit('incoming_message', {
+        data: {
+          id: 'garbage',
+          method: 'error',
+          data: 'someone else\'s error'
+        }
+      })
+      this.plugin.emit('incoming_message', this.response)
+
+      yield promise
+    })
+
+    it('should reject on error message', function * () {
+      this.response.data.method = 'error'
+      this.response.data.data = { message: 'there was an error' }
+
+      const promise = ILQP._sendAndReceiveMessage(this.params)
+      this.plugin.emit('incoming_message', this.response)
+
+      yield expect(promise)
+        .to.eventually.be.rejectedWith(/there was an error/)
+    })
+
+    it('should time out without response', function * () {
```

Comment:
the timeout is configurable, so I can set it to something like 10 milliseconds. What's the reasoning for using fake timers?